addons/hw_drivers/iot_handlers/drivers/printer_driver_base.py

```diff
@@ -199,6 +202,71 @@ def open_cashbox(self, data):
         for drawer in commands['drawers']:
             self.print_raw(drawer)
 
+    def _read_escpos(self):
+        """Wraps the python-escpos read function to perform repeated reads.
+
+        This is necessary due to an issue when using USB, where it takes several
+        reads to get the answer of the command that has just been sent.
+        """
+        MAX_ATTEMPTS = 5
+        result = None
+        attempt = 0
+
+        while not result and attempt < MAX_ATTEMPTS:
+            result = self.escpos_device._read()
+            time.sleep(0.05)
+            attempt += 1
+            _logger.debug("Read attempt %s: %s", attempt, result)
+
+        return result
+
+    def is_escpos_online(self):
+        self.escpos_device._raw(RT_STATUS_ONLINE)
+        status = self._read_escpos()
+        if not status:
+            # Read failed, assume we can print so as not to block user
+            return True
+
+        return not (status[0] & RT_MASK_ONLINE)
+
+    def escpos_paper_status(self):
+        self.escpos_device._raw(RT_STATUS_PAPER)
+        status = self._read_escpos()
+        if not status:
+            # Read failed, assume we can print so as not to block user
+            return "OK"
+
+        if status[0] & RT_MASK_NOPAPER == RT_MASK_NOPAPER:
+            return "EMPTY"
+        if status[0] & RT_MASK_LOWPAPER == RT_MASK_LOWPAPER:
+            return "LOW"
+        else:
+            return "OK"
+
+    def check_escpos_status(self):
+        if not self.escpos_device:
+            return True
+        try:
+            with EscposIO(self.escpos_device, autocut=False):
+                if self.device_connection == 'direct':
+                    # Ensure the read buffer is clear for USB
+                    self._read_escpos()
+
+                if not self.is_escpos_online():
+                    self.send_status(status='error', message='ERROR_OFFLINE')
+                    return False
+                paper_status = self.escpos_paper_status()
+                if paper_status == "LOW":
+                    self.send_status(status='warning', message='WARNING_LOW_PAPER')
+                if paper_status == "EMPTY":
+                    self.send_status(status='error', message='ERROR_NO_PAPER')
+                    return False
+        except (DeviceNotFoundError, OSError, AssertionError):
```

Comment:
Should we catch again basic escpos exception `escpos.exceptions.Error` or are we sure that only `DeviceNotFoundError` can be triggered here?
https://python-escpos.readthedocs.io/en/latest/_modules/escpos/exceptions.html#Error

```suggestion
        except (escpos.exceptions.Error, OSError, AssertionError):
```